File: Task2/apz-pzpi-21-3-merkulov-arsenii-task2/src/HospitalityHub.Core/DTOs/Stats/OccupancyReportResult.cs
using HospitalityHub.Core.Enums;

namespace HospitalityHub.Core.DTOs.Stats;

public class OccupancyReportResult
{
    public DateTime GenerationDate { get; set; }

    public DateTime DateFrom { get; set; }

    public DateTime DateTo { get; set; }

    public int TotalRooms { get; set; }

    public int TotalBookings { get; set; }

    public decimal TotalIncome { get; set; }

    public int TotalGuests { get; set; }

    public decimal AverageIncomePerBooking { get; set; }

    public decimal AverageGuestsPerBooking { get; set; }

    public decimal AverageIncomePerGuest { get; set; }
    
    public decimal AverageIncomePerDay { get; set; }
    
    public string MostProfitableRoomType { get; set; }
    
    public decimal MostProfitableRoomTypeIncome { get; set; }
}

File: apzkr-pzpi-21-3-merkulov-arsenii/Task3-WebClient/hospitality-hub/src/app/api-proxy/models/occupancy-report-result.ts
/* tslint:disable */
/* eslint-disable */
export interface OccupancyReportResult {
  averageGuestsPerBooking?: number;
  averageIncomePerBooking?: number;
  averageIncomePerDay?: number;
  averageIncomePerGuest?: number;
  dateFrom?: string;
  dateTo?: string;
  generationDate?: string;
  mostProfitableRoomType?: string | null;
  mostProfitableRoomTypeIncome?: number;
  totalBookings?: number;
  totalGuests?: number;
  totalIncome?: number;
  totalRooms?: number;
}
